src/System/Net/HttpListener.cs

```diff
@@ -27,15 +28,32 @@ public sealed unsafe partial class HttpListener : IDisposable
         private AuthenticationSchemes _authenticationScheme = AuthenticationSchemes.Anonymous;
         private ExtendedProtectionSelector? _extendedProtectionSelectorDelegate;
         private string? _realm;
+        private X509Certificate2? _certificate;
 
         internal ICollection PrefixCollection => _uriPrefixes.Keys;
 
-        public HttpListener()
+        public HttpListener(X509Certificate2? certificate = null)
         {
             _state = State.Stopped;
             _internalLock = new object();
             _defaultServiceNames = new ServiceNameStore();
 
+            _certificate = certificate;
+            _timeoutManager = new HttpListenerTimeoutManager(this);
+            _prefixes = new HttpListenerPrefixCollection(this);
+
+            // default: no CBT checks on any platform (appcompat reasons); applies also to PolicyEnforcement
+            // config element
+            _extendedProtectionPolicy = new ExtendedProtectionPolicy(PolicyEnforcement.Never);
+        }
+
+        public HttpListener(string certPemFilePath, string? keyPemFilePath = default)
```

Comment:
Don't copy paste the entire contents of this constructor.

Reply:
I got rid of it, since we can init the X5092Certificate via a constructor.